### PR TITLE
Add manage modal for DateTime values [SCI-4081]

### PR DESCRIPTION
**Pull request by urbanrotnik**
Jira ticket: [SCI-4081](https://biosistemika.atlassian.net/browse/SCI-4081)

### What was done
Mange modal for Date, Time and DateTime

### CODE REVIEW

File: app/assets/javascripts/repository_columns/index.js [hunk 1]
```diff
@@ -175,15 +179,16 @@ var RepositoryColumns = (function() {
           .focus();
         $manageModal
           .trigger('columnModal::partialLoadedForStatuses')
-          .trigger('columnModal::partialLoadedForLists');
+          .trigger('columnModal::partialLoadedForLists')
+          .trigger('columnModal::partialLoadedForDateTime');
```
Comment (aignatov-bio) [inline app/assets/javascripts/repository_columns/index.js]: Maybe only trigger selected element?

File: app/models/repository_column.rb [hunk 2]
```diff
@@ -66,4 +66,8 @@ def update_repository_table_states_with_removed_column
   def importable?
     Extends::REPOSITORY_IMPORTABLE_TYPES.include?(data_type.to_sym)
   end
+
+  def range?
```
Comment (okriuchykhin) [inline app/models/repository_column.rb]: Why is it here?

File: app/models/repository_column.rb [hunk 3]
```diff
@@ -66,4 +66,8 @@ def update_repository_table_states_with_removed_column
...
+    repository_date_time_range_value? || repository_date_range_value? || repository_time_range_value?
```
Comment (okriuchykhin) [inline app/models/repository_column.rb]: :\

File: app/assets/javascripts/repository_columns/columns_initializers/list_column_type.js [hunk 4]
```diff
@@ -112,15 +112,15 @@ var RepositoryListColumnType = (function() {
     $manageModal
       .off('change keyup paste', listItemsTextarea)
       .off('change', dropdownDelimiter)
-      .off('columnModal::partialLoadedForLists')
+      .off('columnModal::partialLoadedForRepositoryListValue')
```
Comment (okriuchykhin) [inline app/assets/javascripts/repository_columns/columns_initializers/list_column_type.js]: Please try to use event delegation here

File: app/views/repository_columns/_manage_column_modal_content.html.erb [hunk 5]
```diff
@@ -34,44 +34,67 @@
...
     </span>
-    <span style="display: none" class="column-type asset-column-type" data-column-type="RepositoryAssetValue">
+    <span style="display: none" class="column-type asset-column-type RepositoryAssetValue">
       <%= render partial: "repository_columns/manage_column_partials/asset.html.erb" %>
     </span>
-    <span style="display: none" class="column-type list-column-type" data-column-type="RepositoryListValue">
+    <span style="display: none" class="column-type list-column-type RepositoryListValue">
       <%= render partial: "repository_columns/manage_column_partials/list.html.erb", locals: { column: @repository_column } %>
     </span>
-    <span style="display: none" class="column-type status-column-type" data-column-type="RepositoryStatusValue">
```
Comment (okriuchykhin) [inline app/views/repository_columns/_manage_column_modal_content.html.erb]: Keep data-column-type for column type

File: spec/controllers/repository_columns/date_time_columns_controller_spec.rb [hunk 6]
```diff
@@ -18,6 +18,7 @@
         repository_id: repository.id,
         repository_column: {
           name: 'name',
+          column_type: 'RepositoryDateTimeValue',
           range: true
```
Comment (okriuchykhin) [inline spec/controllers/repository_columns/date_time_columns_controller_spec.rb]: Is it still used?

Comment (urbanrotnik) [reply spec/controllers/repository_columns/date_time_columns_controller_spec.rb]: yes

File: app/views/repository_columns/manage_column_partials/_datetime.html.erb [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+    column&.repository_date_range_value? ||
+    column&.repository_time_range_value? %>
+
+<div class="form-group">
+  <div class="col-sm-3"></div>
+  <div class="col-sm-9">
+    <div class="permission-selector">
+      <%= check_box_tag 'range', '', range_type, { id: 'datetime-range', class: column.new_record? ? 'simple-checkbox' : 'simple-checkbox disabled' } %>
+      <span class="checkbox-label"></span>
+      <span class="range-label">Range</span>
```
Comment (okriuchykhin) [inline app/views/repository_columns/manage_column_partials/_datetime.html.erb]: Use I18n

File: app/views/repository_columns/_manage_column_modal_content.html.erb [hunk 8]
```diff
@@ -34,27 +34,46 @@
...
                   data-edit-url="<%= repository_repository_columns_asset_column_path(@repository, @repository_column) unless @repository_column.new_record? %>"
                   value="RepositoryAssetValue"
-                  <%= 'selected' if @repository_column.RepositoryAssetValue? %> >
+                  <%= 'selected' if @repository_column.repository_asset_value? %> >
             <%= t('libraries.manange_modal_column.select.repository_asset_value') %>
           </option>
+
+          <option data-create-url="<%= repository_repository_columns_date_time_columns_path(@repository) %>"
+                  data-edit-url="<%= repository_repository_columns_asset_column_path(@repository, @repository_column) unless @repository_column.new_record? %>"
+                  value=<%= %>"RepositoryDateValue"
```
Comment (okriuchykhin) [inline app/views/repository_columns/_manage_column_modal_content.html.erb]: Typo?

**Pull request merged**